python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -144,70 +146,100 @@ def preprocess_ondisk_dataset(input_config_path: str) -> str:
+                intput_set_split, output_set_split
             ):
-                # Always save the feature in numpy format.
-                output_data["format"] = "numpy"
-                output_data["path"] = str(
-                    processed_dir_prefix
-                    / input_data["path"].replace("pt", "npy")
-                )
-                if input_data["format"] == "numpy":
-                    # If the original format is numpy, just copy the file.
-                    os.makedirs(
-                        dataset_path / os.path.dirname(output_data["path"]),
-                        exist_ok=True,
-                    )
-                    shutil.copy(
-                        dataset_path / input_data["path"],
-                        dataset_path / output_data["path"],
-                    )
-                else:
-                    # If the original format is not numpy, convert it to numpy.
-                    input_set = read_data(
-                        dataset_path / input_data["path"],
-                        input_data["format"],
-                    )
-                    save_data(
-                        input_set,
-                        dataset_path / output_data["path"],
-                        output_set_per_type["format"],
+                for input_data, output_data in zip(
+                    input_set_per_type["data"], output_set_per_type["data"]
+                ):
+                    # Always save the feature in numpy format.
+                    output_data["format"] = "numpy"
+                    output_data["path"] = str(
+                        processed_dir_prefix
+                        / input_data["path"].replace("pt", "npy")
                     )
+                    if input_data["format"] == "numpy":
+                        # If the original format is numpy, just copy the
+                        # file.
+                        os.makedirs(
+                            os.path.dirname(output_data["path"]),
+                            exist_ok=True,
+                        )
+                        shutil.copyfile(
+                            input_data["path"],
+                            output_data["path"],
+                        )
+                    else:
+                        # If the original format is not numpy, convert it
+                        # to numpy.
+                        input_set = read_data(
+                            input_data["path"],
+                            input_data["format"],
+                        )
+                        save_data(
+                            input_set,
+                            output_data["path"],
+                            output_set_per_type["format"],
+                        )
 
     # 8. Save the output_config.
-    output_config_path = dataset_path / "output_config.yaml"
+    output_config_path = processed_dir_prefix / "metadata.yaml"
```

Comment:
as this path is also checked at the top to see if already preprocessed, is it better to make is as constant variable instead of literal strings.